test/Streamly/Test/Data/Parser.hs

```diff
@@ -372,6 +372,25 @@ takeWhile1 =
         where
             predicate = (== 0)
 
+choice :: Property
+choice =
```

Comment:
Can we test `asum` as well using this same test?

Reply:
Using the same test failed, so probably not. I have not looked into why it failed.

Reply:
That means there is something wrong with this or with asum. We should file an issue if we are not investigating it right now.

Reply:
See https://github.com/composewell/streamly/issues/1191